### ISSUE TITLE
Custom component wrapped by chakra factory cannot be used as a PopoverToggle child

### ISSUE THREAD

**Issue by pribilinskiy** [chakra-ui/chakra-ui#4816]
### Description

Custom component wrapped with `chakra` factory throws an error when used within `PopoverToggle`

### Link to Reproduction

https://codesandbox.io/s/custom-component-wrapped-with-chakra-factory-issue-with-popover-k5ups

### Steps to reproduce

1. Custom component wrapped with `chakra` factory has an issue with `PopoverToggle`

```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
```

2. Notice a Warning from Popper.js (it appears even with a plain non-wrapped `Button`)

```
Popper: CSS "margin" styles cannot be used to apply padding between the popper and its reference element or boundary. To replicate margin, use the `offset` modifier, as well as the `padding` option in the `preventOverflow` and `flip` modifiers. 
```

3. Obviously because the `ref` is not provided by `chakra` factory the popover is placed in a wrong place

4. Problem is solved by wrapping the component with `forwardRef` and passing that to `chakra` factory (see `<XButtonWithRef>`) but that introduces another problem with Typescript typings in that we are losing autocompletion suggestions in IDE (in Codesandbox too)

![image](https://user-images.githubusercontent.com/3078595/136671752-8c92ca99-e90f-47be-942f-c6b07cb29fd2.png)
![image](https://user-images.githubusercontent.com/3078595/136671820-59af042c-5afb-4772-ae48-d0b952fce55a.png)

Please open the console to see the error and warning:
![image](https://user-images.githubusercontent.com/3078595/136671515-a8b846df-ad80-4af6-9fcc-296b6c47f1ac.png)

### Chakra UI Version

1.6.8

### Operating System

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by pribilinskiy** [comment]
Ok, [in the docs](https://chakra-ui.com/docs/overlay/popover#basic-usage) it is already said that **it should take a ref**, although perhaps this could be avoided by using `React.Children(React.cloneElement` and attaching the `onClick` prop, isn't it?

But then the [Chakra factory](https://chakra-ui.com/docs/features/chakra-factory) does not mention the case with using `forwardRef` so that a wrapped element could be provided as children to other elements like `PopoverToggle` or `MenuButton`

**Comment by lenybernard** [comment]
Indeed, to use a custom comopnent, we have to wrap it under the chakra's forwardRef to get it working : 

```typescript
//...somewhere
<Popover
    initialFocusRef={initialFocusRef}
    onClose={onFormSubmit}
    placement="bottom"
>
    <PopoverTrigger>
        <MyButton>Open</MyButton>
    </PopoverTrigger>
    ...
```

```typescript
// src/components/button.tsx
import { Button , Box, forwardRef } from '@chakra-ui/react'

const MyButton = forwardRef(({ someProperty }: MyButtonProps, ref) => {
    return <Button ... ref={ref}>Click</Button>
```
